src/aws/osml/tile_server/ogc/common_geodata.py

```diff
@@ -0,0 +1,13 @@
+#  Copyright 2024 Amazon.com, Inc. or its affiliates.
+from enum import Enum
+
+
+class DataType(str, Enum):
```

Comment:
We have StringEnum class which auto converts the variable into string. 

```
from aws.osml.tile_server.utils import AutoLowerStringEnum

class DataType(str, AutoLowerStringEnum):
  MAP = auto()
  VECTOR = auto() ...
```

Similar to how this is setup: https://github.com/aws-solutions-library-samples/osml-tile-server/blob/main/src/aws/osml/tile_server/viewpoint/models.py#L15-L40

Reply:
I'm not a big fan of those classes and don't suggest using them here. Here's my rationale:
1. In this case an auto() enum isn't appropriate. The values must be "map", "vector", "coverage", or other TBD values that are defined in the spec. I don't want some future developer coming in and changing MAP to MAP_VALUE and then suddenly the API is broken because that isn't an OGC defined data type.
2. When you take module dependencies into consideration depending on an OSML defined AutoLowerStringEnum adds more complexity with very little value. As it stands the `ogc` module is standalone and it only depends on Pydantic. In the future we may break this module out or replace it with some 3rd party implementation of the API. If it starts taking dependencies on other OSML modules that becomes harder to do.
3. Python is eventually addressing some of this by [adding StrEnum in Python 3.11](https://docs.python.org/3.11/library/enum.html). If this was really bothering me now we could have added the [3rd party StrEnum library](https://pypi.org/project/StrEnum/) to avoid linkage to an OSML specific utility but that just doesn't seem worth the effort for this one type.

Reply:
Not blocking, but 1. can be addressed with unit testing.  Since it's in 3.11 I think that is a better path than 3rd party and I'm ok if we want to leave it as is for now, although in general I think the auto() approach is preferable.

Reply:
Not blocking, but 1. can be addressed with unit testing.  Since it's in 3.11 I think that is a better path than 3rd party and I'm ok if we want to leave it as is for now, although in general I think the auto() approach is preferable.

Reply:
What problem does auto() solve for us here? I understand that we could write new unit tests to prevent it from assigning a value that is incompatible with the OGC API but it seems easier and more direct to just create the Enum with the actual mandated values. 

Is there some best practice justification I'm missing here?